### PR TITLE
Switch the ADAL android to Android Studio

### PR DESCRIPTION
**Pull request by heidijinxujia**
Configure the gradle build for the Android Studio.

### CODE REVIEW

**Comment by weijjia** [comment]
It looks good to me. :shipit:

File: adal/build.gradle [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                def fileName = "${archivesBaseName}-${version}.aar"
+                output.outputFile = new File(outputFile.parent, fileName)
+            }
+        }
+    }
+
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (weijjia) [inline adal/build.gradle]: nit: can we reorder it, and put comment 
/*Compile Dependencies*/
compile dependency
/*Test Dependency*/
test dependency

File: sample/build.gradle [hunk 2]
```diff
@@ -0,0 +1,32 @@
+apply plugin: 'com.android.application'
+
+android {
+    compileSdkVersion 21
+    buildToolsVersion "21.1.1"
+    //useLibrary 'org.apache.http.legacy' //FOR sdk 23
```
Comment (weijjia) [inline sample/build.gradle]: can we remove it if it's commented out?

File: sample/build.gradle [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    defaultConfig {
+        applicationId "com.microsoft.aad.adal.sample"
+        minSdkVersion 14
+        targetSdkVersion 21
+        versionCode 1
+        versionName "1.0"
+    }
+    buildTypes {
```
Comment (weijjia) [inline sample/build.gradle]: nit: extra line

File: testapp/proguard-rules.pro [hunk 4]
```diff
@@ -1,11 +1,8 @@
...
-# to define the proguard.config property as described in that file.
-#
 # Add project specific ProGuard rules here.
 # By default, the flags in this file are appended to flags specified
-# in ${sdk.dir}/tools/proguard/proguard-android.txt
-# You can edit the include path and order by changing the ProGuard
-# include property in project.properties.
+# in /Users/randallilama/sdk/android-sdk-macosx/tools/proguard/proguard-android.txt
+# You can edit the include path and order by changing the proguardFiles
+# directive in build.gradle.
```
Comment (addev-ashish) [inline testapp/proguard-rules.pro]: shouldn't we just delete this?

File: sample/build.gradle [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        }
+    }
+
+    lintOptions {
+        abortOnError false
+    }
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (weijjia) [inline sample/build.gradle]: nit: same as above
/*Compile Dependency*/
compile dependency
/*Test dependency*/
test dependency

File: testapp/build.gradle [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        }
+    }
+
+    lintOptions {
+        abortOnError false
+    }
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (weijjia) [inline testapp/build.gradle]: nit: same as above :)

**Comment by weijjia** [comment]
Looks good to me @heidijinxujia 
Only left some minor comments. 
:shipit:  when those minor comments are addressed.

**Comment by weijjia** [comment]
Also, the travis build is failing. why? is it because the travis.yml we use now? but this branch should already has the updated travis script.

**Comment by heidijinxujia** [comment]
@weijjia  It is the travis config for AS. I made a PR before and worked fine. I am not sure why it failed. Working on it now.

File: userappwithbroker/build.gradle [hunk 7]
```diff
@@ -0,0 +1,23 @@
+apply plugin: 'com.android.application'
+
+android {
+    compileSdkVersion 21
+    buildToolsVersion "23.0.3"
```
Comment (weijjia) [inline userappwithbroker/build.gradle]: :) shouldn't be 23, we should only update it when sdk version gets bumped to 23.

**Comment by addev-ashish** [comment]
Take care of Wei's comments and then i'm good.

**Comment by addev-ashish** [comment]
:shipit:

File: adal/build.gradle (hunk 1 above)
Comment (heidijinxujia) [inline adal/build.gradle]: addressed

File: adal/proguard-rules.pro [hunk 8]
```diff
@@ -1,11 +1,8 @@
-# To enable ProGuard in your project, edit project.properties
-# to define the proguard.config property as described in that file.
-#
 # Add project specific ProGuard rules here.
 # By default, the flags in this file are appended to flags specified
-# in ${sdk.dir}/tools/proguard/proguard-android.txt
-# You can edit the include path and order by changing the ProGuard
-# include property in project.properties.
+# in /Users/randallilama/sdk/android-sdk-macosx/tools/proguard/proguard-android.txt
```
Comment (addev-ashish) [inline adal/proguard-rules.pro]: Can you please change "**/Users/randallilama/sdk/android-sdk-macosx/tools/proguard/proguard-android.txt**" to **${sdk.dir}/tools/proguard/proguard-android.txt**

File: sample/build.gradle (hunk 2 above)
Comment (addev-ashish) [inline sample/build.gradle]: did you miss this one?

File: sample/proguard-rules.pro [hunk 9]
```diff
@@ -1,11 +1,8 @@
-# To enable ProGuard in your project, edit project.properties
-# to define the proguard.config property as described in that file.
-#
 # Add project specific ProGuard rules here.
 # By default, the flags in this file are appended to flags specified
-# in ${sdk.dir}/tools/proguard/proguard-android.txt
-# You can edit the include path and order by changing the ProGuard
-# include property in project.properties.
+# in /Users/randallilama/sdk/android-sdk-macosx/tools/proguard/proguard-android.txt
```
Comment (addev-ashish) [inline sample/proguard-rules.pro]: same comment as previous proguard-rules file

File: testapp/proguard-rules.pro (hunk 4 above)
Comment (addev-ashish) [inline testapp/proguard-rules.pro]: or atleast rename the path, same comment as earlier

File: userappwithbroker/build.gradle (hunk 7 above)
Comment (addev-ashish) [inline userappwithbroker/build.gradle]: did you miss this one?

**Pull request merged**